c7n/resources/batch.py

```diff
@@ -41,3 +43,109 @@ class resource_type(object):
+            self.__class__.__name__, len(results), orig_length, key.lower()))
+        return results
+
+
+@ComputeEnvironment.action_registry.register('update-environment')
+class UpdateComputeEnvironment(BaseAction, StateTransitionFilter):
+    """Updates an AWS batch compute environment
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: update-environments
+            resource: batch-compute
+            filters:
+              - computeResources.desiredvCpus: 0
+              - state: ENABLED
+            actions:
+              - type: update-environment
+                state: DISABLED
+    """
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['update-environment']},
+            'computeEnvironment': {'type': 'string'},
+            'state': {'type': 'string', 'enum': ['ENABLED', 'DISABLED']},
+            'computeResources': {
+                'type': 'object',
+                'additionalProperties': False,
+                'properties': {
+                    'minvCpus': {'type': 'integer'},
+                    'maxvCpus': {'type': 'integer'},
+                    'desiredvCpus': {'type': 'integer'}
+                }
+            },
+            'serviceRole': {'type': 'string'}
+        }
+    }
+    permissions = ('batch:UpdateComputeEnvironment',)
+    valid_origin_status = ('VALID', 'INVALID')
+
+    def process(self, resources):
+        resources = self.filter_resource_state(
+            resources, 'status', self.valid_origin_status)
+        client = local_session(self.manager.session_factory).client('batch')
+        params = dict(self.data)
+        params.pop('type')
+        for r in resources:
+            params['computeEnvironment'] = r['computeEnvironmentName']
+            client.update_compute_environment(**params)
+
+
+@ComputeEnvironment.action_registry.register('delete')
+class DeleteComputeEnvironment(BaseAction, StateTransitionFilter):
+    """Delete an AWS batch compute environment
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: delete-environments
+            resource: batch-compute
+            filters:
+              - computeResources.desiredvCpus: 0
+            action:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    permissions = ('batch:DeleteComputeEnvironment',)
+    valid_origin_states = ('DISABLED',)
+    valid_origin_status = ('VALID', 'INVALID')
+
+    def delete_environment(self, r):
+        client = local_session(self.manager.session_factory).client('batch')
+        client.delete_compute_environment(
+            computeEnvironment=r['computeEnvironmentName'])
+
+    def process(self, resources):
+        resources = self.filter_resource_state(
+            self.filter_resource_state(
+                resources, 'state', self.valid_origin_states),
+            'status', self.valid_origin_status)
```

Comment:
same here wrt if we filter all the resources (ie if not resources: ) it would be good to log to that effect that we're implicitly doing filter in the action.